R/data_modify.R

```diff
@@ -26,6 +26,14 @@
 #' Note that newly created variables can be used in subsequent expressions.
 #' See also 'Examples'.
 #'
+#' @param .at A character vector of variable names that should be modified. This
+#' argument is used in combination with the `.modify` argument.
+#' @param .if A function that returns `TRUE` for columns in the data frame where
+#' `.if` applies. This argument is used in combination with the `.modify` argument.
+#' @param .modify A function that modifies the variables defined in `.at` or `.if`.
+#' This argument is used in combination with either the `.at` or the `.if`argument.
+#' If `.modify` is not provided, `.at` and `.if` are ignored.
```

Comment:
I think this case should error as it is unexpected usage of `.if` and `.at`.

This should also specify that the output of `.modify` must be either of length 1 or of same length as the input. For example this errors:
```r
d <- iris[1:5, ]
data_modify(d, .if = is.numeric, .modify = function(x) unique(x))
```